Products/ZenModel/UserSettings.py

```diff
@@ -1151,6 +1151,29 @@ def manage_deleteAdministrativeRole(self, delids=(), REQUEST=None):
                 )
             return self.callZenScreen(REQUEST)
 
+    security.declareProtected(ZEN_CHANGE_SETTINGS, 'getAllAdminGuids')
+    def getAllAdminGuids(self, returnChildrenForRootObj=False):
+        """
+        Return all guids user has permissions to.
+        """
+        guids = []
+        ars = self.getAllAdminRoles()
+        for ar in ars:
+            if not returnChildrenForRootObj:
+                guids.append(IGlobalIdentifier(ar.managedObject()).getGUID())
+            else:
+                if ar.managedObject().getPrimaryId() in (
+                        '/zport/dmd/Devices',
+                        '/zport/dmd/Locations',
+                        '/zport/dmd/Groups',
+                        '/zport/dmd/Systems'
+                ):
```

Comment:
The tuple of organizer names should be kept in a variable with an informative name, e.g.
```
    rootOrganizers = (
        '/zport/dmd/Devices',
        ...
    )
    if ar.managedObject().getPrimaryId() in rootOrganizers:
        ...
```